src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Analysis.cs

```diff
@@ -0,0 +1,98 @@
+            if (analyzeSemanticModel != null) { context.RegisterSemanticModelAction(c => AnalyzeIfEnabled(c, analyzeSemanticModel)); }
+            if (analyzeSyntaxTree != null) { context.RegisterSyntaxTreeAction(c => AnalyzeIfEnabled(c, analyzeSyntaxTree)); }
+            if (analyzeSyntax != null) { context.RegisterSyntaxNodeAction(c => AnalyzeIfEnabled(c, analyzeSyntax), syntaxKinds); }
+            if (analyzeOperation != null) { context.RegisterOperationAction(c => AnalyzeIfEnabled(c, analyzeOperation), operationKinds); }
+
+            // check for all the other things that can be registered as well.
+        }
+
+        private void AnalyzeIfEnabled(CodeBlockAnalysisContext context, Action<CodeBlockAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            => AnalyzeIfEnabled(context, analyze, context.Options, context.SemanticModel.SyntaxTree, context.CancellationToken);
+
+        private void AnalyzeIfEnabled(OperationAnalysisContext context, Action<OperationAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            => AnalyzeIfEnabled(context, analyze, context.Options, context.Operation.SemanticModel.SyntaxTree, context.CancellationToken);
+
+        private void AnalyzeIfEnabled(SemanticModelAnalysisContext context, Action<SemanticModelAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            => AnalyzeIfEnabled(context, analyze, context.Options, context.SemanticModel.SyntaxTree, context.CancellationToken);
+
+        private void AnalyzeIfEnabled(SyntaxNodeAnalysisContext context, Action<SyntaxNodeAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            => AnalyzeIfEnabled(context, analyze, context.Options, context.SemanticModel.SyntaxTree, context.CancellationToken);
+
+        private void AnalyzeIfEnabled(SyntaxTreeAnalysisContext context, Action<SyntaxTreeAnalysisContext, CodeStyleOption<TOptionKind>> analyze)
+            => AnalyzeIfEnabled(context, analyze, context.Options, context.Tree, context.CancellationToken);
+
+        private void AnalyzeIfEnabled<TContext>(
+            TContext context, Action<TContext, CodeStyleOption<TOptionKind>> analyze,
+            AnalyzerOptions options, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var optionValue = optionSet.GetOption(_option);
+            switch (optionValue.Notification.Severity)
+            {
+                case ReportDiagnostic.Error:
+                case ReportDiagnostic.Warn:
+                case ReportDiagnostic.Info:
+                    break;
+                default:
+                    // don't analyze if it's any other value.
+                    return;
+            }
```

Comment:
@sharwell tehre are 6 values for this enum.

Default, Error, Warn, Info, Hidden, Suppress.

Error/Warn/Info are all things that should be checked by the analyzer.

'Hidden' is the same as 'Refactoring' and should not be checked by teh analyzer.

I'm pretty sure 'Suppress' should not be checked by the analyzer.

This only leaves 'Default'. I'm not sure waht that value means and if the analyzer should run when in that state or not.